### PR TITLE
[MM-52835] : Migrate "components/admin_console/settings_group.jsx" and tests to Typescript

### PR DESCRIPTION
**Pull request by tejaskarelia17**
#### Summary
In this PR, completes migration of `components/admin_console/settings_group.jsx` to TypeScript.

#### Ticket Link
Fixes [GitHub issue #23425](https://github.com/mattermost/mattermost-server/issues/23425)
https://mattermost.atlassian.net/browse/MM-52835

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: webapp/channels/src/components/admin_console/settings_group.tsx [hunk 1]
```diff
@@ -4,26 +4,33 @@
...
+export default class SettingsGroup extends React.PureComponent<Props> {
+    static propTypes = {
+        id: PropTypes.string,
+        show: PropTypes.bool.isRequired,
+        header: PropTypes.node,
+        title: PropTypes.node,
+        subtitle: PropTypes.node,
+        children: PropTypes.node,
+        container: PropTypes.bool,
+    };
```
Comment (larkox) [inline webapp/channels/src/components/admin_console/settings_group.tsx:26]: This is no longer needed

File: webapp/channels/src/components/admin_console/settings_group.tsx [hunk 2]
```diff
@@ -4,26 +4,33 @@
...
 
-    render() {
+    static defaultProps = {
+        show: true,
+        container: true,
+    };
```
Comment (larkox) [inline webapp/channels/src/components/admin_console/settings_group.tsx:31]: This shouldn't be needed either:
https://react-typescript-cheatsheet.netlify.app/docs/basic/getting-started/default_props/#you-may-not-need-defaultprops

**Review by larkox — CHANGES_REQUESTED** [review]
Thanks for the contribution! Just two minor things.

**Review by crspeller — APPROVED** [review]
Looks good past @larkox feedback!

File: webapp/channels/src/components/admin_console/settings_group.tsx (hunk 2 above)
Comment (tejaskarelia17) [reply webapp/channels/src/components/admin_console/settings_group.tsx]: Thanks @larkox for the reference article. I updated the code and passed the required prop `show` from the two components that were initially relying on the `defaultProps` from `SettingsGroup`. Let me know if this is the correct approach.

**Review by larkox — CHANGES_REQUESTED** [review]
Thanks for the changes. Just one minor clarification.

**Review by yasserfaraazkhan — APPROVED** [review]

**Pull request merged**